src/App/Fossa/VPS/Scan.hs

```diff
@@ -0,0 +1,140 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.VPS.Scan
+  ( scanMain,
+    SkipIPRScan (..)
+  ) where
+
+import Control.Effect.Lift (Lift, sendIO)
+import Control.Carrier.Diagnostics
+import Effect.Exec
+import System.Exit (exitFailure)
+import Control.Concurrent.Async (concurrently)
+
+import App.Fossa.VPS.EmbeddedBinary
+import App.Fossa.VPS.Scan.Core
+import App.Fossa.VPS.Scan.RunIPR
+import App.Fossa.VPS.Scan.RunSherlock
+import App.Fossa.VPS.Scan.ScotlandYard
+import App.Fossa.VPS.Types
+import App.Fossa.ProjectInference
+import App.Types (BaseDir (..), ApiKey (..), OverrideProject (..), ProjectRevision (..))
+import Control.Effect.Exception (bracket)
+import Data.Aeson
+import Data.Text (Text)
+import Effect.Logger
+import Path
+import Text.URI (URI)
+
+newtype SkipIPRScan = SkipIPRScan {unSkipIPRScan :: Bool}
+
+scanMain :: URI -> BaseDir -> ApiKey -> Severity -> OverrideProject -> FilterExpressions -> SkipIPRScan ->  IO ()
+scanMain baseuri basedir apikey logSeverity overrideProject fileFilters skipIprScan = do
+  let fossaOpts = FossaOpts baseuri $ unApiKey apikey
+      partVpsOpts = PartialVPSOpts fossaOpts (unSkipIPRScan skipIprScan) fileFilters
+
+  result <- runDiagnostics $ bracket extractEmbeddedBinaries cleanupExtractedBinaries $ vpsScan basedir logSeverity overrideProject partVpsOpts
```

Comment:
while we're here, can we extract `bracket embeddedBinaries cleanupExtractedBinaries` into a function we export from `App.Fossa.VPS.EmbeddedBinary`?